pseudomonas_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -304,6 +311,132 @@
+
+
 # ### Rank pathways 
 
+# In[27]:
+
+
+# Concatenate simulated experiments
+simulated_GSEA_stats_all = process.concat_simulated_data(local_dir, num_runs, project_id, 'GSEA')
+simulated_GSEA_stats_all.set_index('pathway', inplace=True)
+print(simulated_GSEA_stats_all.shape)
+
+
+# In[28]:
+
+
+# Aggregate statistics across all simulated experiments
+simulated_GSEA_summary_stats = calc.aggregate_stats(
+    col_to_rank_pathways,
+    simulated_GSEA_stats_all,
+    'GSEA'
+)
+
+simulated_GSEA_summary_stats.sort_values(by=('padj', 'median')).head()
+
+
+# In[29]:
+
+
+# Load association statistics for template experiment
+template_GSEA_stats = template_enriched_pathways.iloc[:, :-1]
+template_GSEA_stats.set_index('pathway', inplace=True)
+
+template_GSEA_stats.head()
+
+# Rank genes in template experiment
+template_GSEA_stats = calc.rank_genes_or_pathways(
+    col_to_rank_pathways,
+    template_GSEA_stats,
+    True
+)
+
+
+# In[30]:
+
+
+# Rank genes in simulated experiments
+simulated_GSEA_summary_stats = calc.rank_genes_or_pathways(
+    col_to_rank_pathways,
+    simulated_GSEA_summary_stats,
+    False
+)
+
+
 # ### Pathway summary table
+
+# In[31]:
+
+
+# Create intermediate file: "<local_dir>/gene_summary_table_<col_to_rank_pathways>.tsv"
+summary_pathway_ranks = process.generate_summary_table(
+    template_GSEA_stats,
+    simulated_GSEA_summary_stats,
+    col_to_rank_pathways,
+    local_dir,
+    'pathway',
+    params
+)
+
+summary_pathway_ranks.sort_values(by='Rank (simulated)', ascending=False).head()
+
+
+# In[32]:
+
+
+# Create `pathway_summary_filename`
+summary_pathway_ranks.to_csv(pathway_summary_filename, sep='\t')
+
+
+# **Quick check:**
+# 
+# Looks like Ribosomes are found to be significantly differentiated in the template and across 25 simulated experiments. So this pathway looks to be generic, which is consistent with [Crow. et. al.](https://www.pnas.org/content/116/13/6491.abstract)
```

Comment:
Crow et al. used human data, right? Did you expect to see some similarities in the generic pathways between humans and Pseudomonas?

Maybe ribosomes in particular make sense since the protein sequences are known to be highly conserved between species (so presumably the functions are too)? Just guessing here, but it could be interesting.

Reply:
Yes, crow et. al. findings were based on human data.

This is a good point. I actually forgot that the data sources were different when I looked at the findings. So i think its not so much that the protein has the same function across species, so much as these proteins are found to be DE across a wide context in both humans and pseudomonas. This this [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2679180/) talks about all the extra roles they can play which is consistent with this finding.